Clarify how the ~> version constraint operator works (#31298)
hashicorp/terraform: website/docs/language/providers/requirements.mdx

directory where you'd run `terraform apply` — should also specify the
_maximum_ provider version it is intended to work with, to avoid accidental
upgrades to incompatible new versions. The `~>` operator is a convenient

> shorthand for allowing the rightmost component of a version to increment. In
> the example below, the operator will permit patch releases within a specific
> minor release:

Reviewer: I like this subtle, but helpful clarification. Two things - we try to stay away from positional language (like below) so I did an edit for that. I also updated it a little bit to remove "will". I try to stay away from that as well, because I want to talk about what the example is and what it's doing (let me know if that makes no sense).

Please push back if I've materially changed the meaning! Otherwise, I appreciate this change and approve!

```suggestion
shorthand for allowing the rightmost component of a version to increment. The following example uses the operator to allow only patch releases within a specific minor release:
```
